### ISSUE TITLE
Nightly failure in ipa server  testing_master_testing: freeipa-healthcheck reports CRITICAL

### ISSUE THREAD

**Issue by fcami** [dogtagpki/pki#3578]
[link](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/b5350b7c-cc1a-11eb-aa0f-fa163e7c9f91/) to report

```
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd110:transport.py:557   {
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd110:transport.py:557     "source": "ipahealthcheck.ipa.files",
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd110:transport.py:557     "check": "IPAFileCheck",
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd110:transport.py:557     "result": "CRITICAL",
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd110:transport.py:557     "uuid": "e090d436-7bcb-4865-87b3-daeac39cbf31",
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd110:transport.py:557     "when": "20210613080658Z",
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd110:transport.py:557     "duration": "0.048507",
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd110:transport.py:557     "kw": {
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd110:transport.py:557       "exception": "[Errno 2] No such file or directory: '/var/log/pki/pki-tomcat/kra/selftests.log'"
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd110:transport.py:557     }
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd110:transport.py:557   },
```

Short extract:

 "[Errno 2] No such file or directory: '/var/log/pki/pki-tomcat/kra/selftests.log'"

**Comment by fcami** [comment]
So, to be clear: the question is, why did that log disappear suddenly?

**Comment by edewata** [comment]
@fcami Are you still seeing this issue? @PsOverflow showed me an IPA CI report showing that the IPA healthcheck is passing:
http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/41b87b54-f508-11eb-83a3-fa163e2b0c01/report.html

I'm not sure why the log disappeared. I don't think the code is intentionally removing it. I'm suspecting it might be related to the `selftests.container.order.startup` parameter in the `CS.cfg`. Does your `CS.cfg` have this parameter? Previously if that parameter was missing, it would cause a [problem](https://bugzilla.redhat.com/show_bug.cgi?id=1930586) to `pki-server cert-fix`. A [workaround](https://github.com/dogtagpki/pki/pull/3466) was added to ignore the missing param, but we didn't really investigate why it was missing or try to prevent it from missing again.

**Comment by fcami** [comment]
Hi @edewata the reason why this test pass now is that such missing files are not considered failures any more.

This is what we have in CS.cfg:

# grep selftests.container.order.startup CS.cfg
selftests.container.order.startup=CAPresence:critical, SystemCertsVerification:critical

**Comment by edewata** [comment]
@fcami Right, if a log file just doesn't exist, it should not be considered a failure. There could be a process external to PKI that cleans it up, although I'm not aware of any. However, if a log file disappears (from existing to not existing), that should be investigated, but we need a way to reproduce it. Without a reproducer there is not much we can do. So could we close this ticket for now? If we can reproduce the issue we can reopen the ticket.

**Comment by flo-renaud** [comment]
The test doesn't install the KRA, meaning it's normal that the file /var/log/pki/pki-tomcat/kra/selftests.log is not created. This issue can be ignored (ipa-healthcheck has already been updated to not report missing log files as CRITICAL error)
